### ISSUE TITLE
k8s ingress auth-tls-insecure annotation not working

### ISSUE THREAD

**Issue by jwheeler-gs** [containous/traefik#4836]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [X] No

### What did you do?
I am running traefik in a development environment with https endpoints enabled, using the default traefik SSL certificate.

I set up the following ingress definition:
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    ingress.kubernetes.io/auth-tls-insecure: "true"
    ingress.kubernetes.io/auth-type: forward
    ingress.kubernetes.io/auth-url: https://login.internal.local.mydomain.com/authorize
    ingress.kubernetes.io/is-development: "true"
    kubernetes.io/ingress.class: traefik
  labels:
    traffic-type: external
  name: test
  namespace: test
spec:
  rules:
  - host: test.local.mydomain.com
    http:
      paths:
      - backend:
          serviceName: test
          servicePort: http
        path: /
```

### What did you expect to see?
I expect the forward auth request to succeed even with traefik's default SSL certificate since auth-tls-insecure is set to true.

### What did you see instead?
```
time="2019-05-01T16:33:50Z" level=debug msg="Error calling https://login.internal.local.mydomain.com/authorize. Cause: Get https://login.internal.local.mydomain.com/authorize: x509: certificate is valid for ffe823de1a97908301ae47d48a08dc4a.64aa04c912381e9780ad2cda057623f6.traefik.default, not login.internal.local.mydomain.com"
```

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
1.7.10
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
logLevel = "DEBUG"

defaultEntryPoints = ["http", "https"]

InsecureSkipVerify = true

[entryPoints]
  [entryPoints.http]
  address = ":80"
  compress = true
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
  compress = true
    [entryPoints.https.tls]
  [entryPoints.traefik]
  address = ":8080"

[api]
entrypoint = "traefik"

# Enable Kubernetes Ingress Provider.
[kubernetes]
labelselector = "traffic-type=external"
```

**Comment by tverhoeven** [comment]
I'm running into the same issue with traefik 1.7.9.

Is there a way to provide a CA using a ingress definition? Would this solve the issue?

**Comment by majkrzak** [comment]
@tverhoeven @jwheeler-gs 
```func getForwardAuthConfig(i *extensionsv1beta1.Ingress, k8sClient Client) (*types.Forward, error) {
	authURL := getStringValue(i.Annotations, annotationKubernetesAuthForwardURL, "")
	if len(authURL) == 0 {
		return nil, fmt.Errorf("forward authentication requires a url")
	}

	forwardAuth := &types.Forward{
		Address:             authURL,
		TrustForwardHeader:  getBoolValue(i.Annotations, annotationKubernetesAuthForwardTrustHeaders, false),
		AuthResponseHeaders: getSliceStringValue(i.Annotations, annotationKubernetesAuthForwardResponseHeaders),
	}

	authSecretName := getStringValue(i.Annotations, annotationKubernetesAuthForwardTLSSecret, "")
	if len(authSecretName) > 0 {
		authSecretCert, authSecretKey, err := loadAuthTLSSecret(i.Namespace, authSecretName, k8sClient)
		if err != nil {
			return nil, fmt.Errorf("failed to load auth secret: %s", err)
		}

		forwardAuth.TLS = &types.ClientTLS{
			Cert:               authSecretCert,
			Key:                authSecretKey,
			InsecureSkipVerify: getBoolValue(i.Annotations, annotationKubernetesAuthForwardTLSInsecure, false),
		}
	}

	return forwardAuth, nil
}
```
This is the code responsible for handling forwarded auth. It's quality is really low. I'm going to fix it tomorrow. Let's hope it will be merged asap.

**Comment by dduportal** [comment]
Closed by #5319 and available in [Traefik `v1.7.15`](https://github.com/containous/traefik/releases/tag/v1.7.15).
